src/comms/config_layer.cpp

```diff
@@ -42,17 +54,96 @@ const Config* const ConfigLayer::configure(HIDLayer* comms) {
     }
 #endif
 
-// update stored config, msg if successful
+    // update stored config, msg if successful
     Serial.println("Attempting to store config...");
-    if (store_config()) Serial.println("Config successfully stored in /config.pack");
-    else Serial.println("Config not successfully stored (is an SD card inserted?)");    // not a fatal error, can still return
+    if (store_config()) Serial.println("\tConfig successfully stored in /config.pack\n");
+    else Serial.println("\tConfig not successfully stored (is an SD card inserted?)\n");    // not a fatal error, can still return
+
+    // blink 4 times quickly to show that we received a config packet finished processing it
+    for (int i = 0; i < 8; i++) {
+        digitalToggle(13);
+        delay(100);
+    }
+
+    // get the outgoing packet and set it to 0
+    CommsPacket* outgoing = comms->get_outgoing_packet();
+    memset(outgoing->raw, 0, sizeof(outgoing->raw));
+
+    // Hive marks the config process as done once teensy sends a packet with info_bit == 0
+    // Hive then sends a packet back with info_bit == 0 as well, so ping until we get a non-config back
+    while (comms->get_incoming_packet()->raw[3] == 1) {
+        comms->ping();
+    }
 
     return &config;
 }
 
+void ConfigLayer::reconfigure(HIDLayer* comms) {
```

Comment:
configure and reconfigure can't have this much duplicate code between them for maintainability reasons. I'd say either make everything just call the one `configure` and make them do the same thing, or have reconfigure call the configure function inside of it, so you can have your before and after behavior still there (which seems to be the only difference).